### PR TITLE
MOB-428 : add custom amount to price triggers screen

### PR DESCRIPTION
**Pull request by mike-dydx**
## Links (dYdX Internal Use Only)
Linear Ticket: [MOB-428 : add custom amount to price triggers screen](https://linear.app/dydx/issue/MOB-428/add-custom-amount-to-price-triggers-screen)

Figma Design: https://www.figma.com/file/mKevZOfE9nj6MZpiolKYW1/dYdX-%E2%80%BA-Mobile?type=design&node-id=5621-14695&mode=design&t=eTjSvS2SdkstNDMd-4

<br/>

## Description / Intuition
- custom amount view
- when on, display custom amount input with minimum amount prefilled
- when modified, modify the triggersInput `size` field

<br/>

## Before/After Screenshots or Videos

| Before | After |
|--------|-------|
| <video src=""> | <video src="https://github.com/dydxprotocol/v4-native-ios/assets/149746839/f9bcf55b-9819-4578-8546-82da782dfb11"> |
| <video src=""> | <video src="https://github.com/dydxprotocol/v4-native-ios/assets/149746839/32bd585a-0dd5-45cc-b7d8-f161f50fdf4d"> |

<br/>

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring or Technical Debt
- [ ] Documentation update
- [ ] Other (please describe: )

### CODE REVIEW

File: dydx/dydx.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "originHash" : "e085067eee7983e4ba42c06fd96e9ea3884445575db37e125bed61bde5eeb3f1",
+  "originHash" : "3cd7346cace16cf660f9c22302c5ca6770335c67902f3fd155011356a9d43929",
```
Comment (mike-dydx) [inline dydx/dydx.xcworkspace/xcshareddata/swiftpm/Package.resolved]: @ruixhuang any chance you know why my package resolved file just keeps updating? It's all over the place

File: PlatformUI/PlatformUI/Components/Input/PlatformInput.swift [hunk 2]
```diff
@@ -524,7 +524,7 @@ open class PlatformBooleanInputViewModel: PlatformValueInputViewModel {
         if let label = label, label.length > 0 {
             return Text(label)
                 .themeColor(foreground: isEnabled ? .textSecondary : .textTertiary)
-                        .themeFont(fontSize: .smaller)
+                        .themeFont(fontSize: .medium)
```
Comment (mike-dydx) [inline PlatformUI/PlatformUI/Components/Input/PlatformInput.swift:527]: tested this in other places (trade input view)

File: dydx/dydx.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (ruixhuang) [reply dydx/dydx.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Yeah.. it happens to me, I'd just revert it without checking it in.

Comment (mike-dydx) [reply dydx/dydx.xcworkspace/xcshareddata/swiftpm/Package.resolved]: i might [remove](https://forums.swift.org/t/package-resolved-should-go-in-the-gitignore/14699)

File: dydx/dydxViews/dydxViews/_v4/TakeProfitStopLoss/Components/dydxCustomAmountViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                    }
+                }
+            }
+            .wrappedInAnyView()
+        }
+    }
+
+    @objc private func sliderValueChanged(sender: UISlider) {
+        guard let stepSizeDecimals else { return }
+        let roundedValue = dydxFormatter.shared.raw(number: NSNumber(value: sender.value), digits: stepSizeDecimals)
```
Comment (ruixhuang) [inline dydx/dydxViews/dydxViews/_v4/TakeProfitStopLoss/Components/dydxCustomAmountViewModel.swift]: Markets like Shiba have stepSize like 100, 1000, 10000, so we'd need to round the size with something like

`size =  Int(size / stepSize) * stepSize`

File: dydx/dydxViews/dydxViews/_v4/TakeProfitStopLoss/Components/dydxCustomAmountViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+            self,
+            action: #selector(self.sliderValueChanged),
+            for: .valueChanged
+        )
+
+        return slider
+    }()
+
+    private var sliderBackground: some View {
+        GeometryReader { geometry in
```
Comment (ruixhuang) [inline dydx/dydxViews/dydxViews/_v4/TakeProfitStopLoss/Components/dydxCustomAmountViewModel.swift:81]: 👍  Nice trick!

File: dydx/dydxViews/dydxViews/_v4/TakeProfitStopLoss/Components/dydxCustomAmountViewModel.swift (hunk 3 above)
Comment (mike-dydx) [reply dydx/dydxViews/dydxViews/_v4/TakeProfitStopLoss/Components/dydxCustomAmountViewModel.swift]: ah, i thought the raw function did that, looks like it does in some implementations, I will update in this PR

**Review by ruixhuang — APPROVED** [review]

**Pull request merged**